lib/src/value/calculation.dart

```diff
@@ -158,6 +381,212 @@ class SassCalculation extends Value {
+  /// arguments is an unquoted `var()` string.
+  static Value rem(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber || modulus is! SassNumber) {
+      return SassCalculation._("rem", args);
+    }
+    _verifyCompatibleNumbers(args);
+
+    var result = dividend.modulo(modulus);
+    if ((modulus.value < 0 && dividend.value >= 0) ||
+        (modulus.value >= 0 && dividend.value < 0)) {
+      if (modulus.value.isInfinite) return dividend;
+      if (result is SassNumber && result.value == 0) return SassNumber(-0.0);
+      return result.minus(dividend);
+    }
+    return result;
+  }
+
+  /// Creates a `mod()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber || modulus is! SassNumber) {
+      return SassCalculation._("mod", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    numberOrStep = numberOrStep.andThen(_simplify);
+    step = step.andThen(_simplify);
+    strategyOrNumber = _simplify(strategyOrNumber);
+
+    var args = [
+      strategyOrNumber,
+      if (numberOrStep != null) numberOrStep,
+      if (step != null) step
+    ];
+
+    switch (args.length) {
```

Comment:
Using a switch for this feels off to me, but maybe this is actually fine given its newfound importance in Dart 3? @nex3, what are your thoughts?

Reply:
It seemed like the most `readable` way to me, since the functionality actually depends on the number of arguments. 🤔

Reply:
I think a switch could make sense, but if you're going to do that I think it would make more sense to match on the tuple `(_simplify(arg1), arg2.andThen(_simplify), arg3.andThen(_simplify))` which should allow you to pull out and simplify a bunch of the inner conditions.